### PR TITLE
Logstream: ensure all failed targets are marked as such

### PR DESCRIPTION
**Pull request by mikejholly**
A missing func call was leading to some target statuses not being updated.

### CODE REVIEW

File: earthfile2llb/converter.go [hunk 1]
```diff
@@ -1862,6 +1862,7 @@ func (c *Converter) FinalizeStates(ctx context.Context) (*states.MultiTarget, er
 		if c.ftrs.ExecAfterParallel {
 			err = c.forceExecution(ctx, c.mts.Final.MainState, c.mts.Final.PlatformResolver)
 			if err != nil {
+				c.RecordTargetFailure(ctx, err)
```
Comment (vladaionescu) [inline earthfile2llb/converter.go:1865]: Will this still work even if the ctx is canceled?

**Review by vladaionescu — APPROVED** [review]

File: earthfile2llb/converter.go (hunk 1 above)
Comment (mikejholly) [reply earthfile2llb/converter.go:1865]: AFAICT, most cancellations are being caught [here](https://github.com/earthly/earthly/blob/main/earthfile2llb/interpreter.go#L74). But some weren't as there is a change the cancellation may trigger a failure in [`FinalizeStates`](https://github.com/earthly/earthly/blob/main/earthfile2llb/earthfile2llb.go#L351-L359) as well. Maybe there's a better spot to handle all cases.

Comment (vladaionescu) [reply earthfile2llb/converter.go:1865]: Perhaps logstream should generally work despite any regular cancelation, with the exception being if the cancelation was user-issued.

Comment (mikejholly) [reply earthfile2llb/converter.go:1865]: That's the idea. With this change and the corresponding backend change, all messages should make their way to the server except after the 2nd control-c. This particular change is to fix incorrect "in progress" statuses in the database &, subsequently, the UI.

Comment (vladaionescu) [reply earthfile2llb/converter.go:1865]: SGTM!

Comment (mikejholly) [reply earthfile2llb/converter.go:1865]: I played around with a few different locations for `RecordTargetFailure` but they didn't seem to catch all cancelations. It seems like the code in this PR is working well.

**Pull request merged**